### ISSUE TITLE
li[2] and li[3] numerical evaluation fails for complex not in rectangular form

### ISSUE THREAD

**Issue by rtoy** [rtoy/maxima#2670]
Imported from SourceForge on 2024-07-06 17:14:51
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-04-17 20:28:22
Original: https://sourceforge.net/p/maxima/bugs/3422

---

`li[2]` and `li[3]` cause a Lisp error for numerical evaluation of complexes which are not in rectangular form.
~~~~
(%i1) li[2]((1 + %i)/2e0);

BIGFLOAT: unable to convert 0.5 (%i + 1.0) to a CL or BIGFLOAT number.
 -- an error. To debug this try: debugmode(true);
(%i2) li[2]((1 + %i)/2b0);

BIGFLOAT: unable to convert 5.0b-1 (%i + 1.0b0) to a CL or BIGFLOAT number.
 -- an error. To debug this try: debugmode(true);
(%i3) li[3]((1 + %i)/2e0);

BIGFLOAT: unable to convert 0.5 (%i + 1.0) to a CL or BIGFLOAT number.
 -- an error. To debug this try: debugmode(true);
(%i4) li[3]((1 + %i)/2b0);

BIGFLOAT: unable to convert 5.0b-1 (%i + 1.0b0) to a CL or BIGFLOAT number.
 -- an error. To debug this try: debugmode(true);
~~~~

This is a bug in `li2numer` and `li3numer`, which need to call `$RECTFORM`. I'm working on a patch.

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:14:53
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-04-17 21:28:04
Original: https://sourceforge.net/p/maxima/bugs/3422/#5cda

---

- **status**: open --> closed

**Comment by rtoy** [comment]
Imported from SourceForge on 2024-07-06 17:14:56
Created by **[robert_dodier](https://sourceforge.net/u/robert_dodier/)** on 2018-04-17 21:28:05
Original: https://sourceforge.net/p/maxima/bugs/3422/#a51e

---

Fixed by commit 4c4a5380. Closing this report.
